### PR TITLE
Fix LTO

### LINKED ISSUE #881: LLVMgold not available on pella, configuration fails
pella has cmake 3.10, thus the build script tries to enable LTO in Release builds - but the check for this fails as LLVMgold is not installed. Should LLVMgold be in the install script?

### PR DESCRIPTION
**Pull request by mrks**
fixes #881

### CODE REVIEW

**Comment by mrks** [comment]
@FabianWiebe Sorry to bug you again - any idea what's going wrong here?

**Comment by mrks** [comment]
Mmmh... that is somewhat disappointing:

```
 Benchmark | prev. iter/s | new iter/s | change
-----------+--------------+------------+--------------
 TPC-H 1   | 0.04         | 0.04       | +0%
 TPC-H 3   | 0.795        | 0.802      | +1%
 TPC-H 5   | 0.021        | 0.021      | +1%
 TPC-H 6   | 2.887        | 2.907      | +1%
 TPC-H 7   | 0.088        | 0.087      | -1%
 TPC-H 9   | 0.017        | 0.017      | -2%
 TPC-H 10  | 0.576        | 0.578      | +0%
 TPC-H 13  | 0.545        | 0.541      | -1%
 average   |              |            | -0%
```

**Comment by mrks** [comment]
Clang:

```
Markus.Dreseler@pella:~/hyrise$ ./scripts/compare_benchmarks.py nolto.json lto.json

 Benchmark | prev. iter/s | new iter/s | change
-----------+--------------+------------+---------------
 TPC-H 1   | 0.034        | 0.036      | +9%
 TPC-H 3   | 0.718        | 0.8        | +11%
 TPC-H 5   | 0.022        | 0.022      | +0%
 TPC-H 6   | 3.079        | 3.166      | +3%
 TPC-H 7   | 0.093        | 0.094      | +1%
 TPC-H 9   | 0.018        | 0.018      | +1%
 TPC-H 10  | 0.643        | 0.674      | +5%
 TPC-H 13  | 0.53         | 0.55       | +4%
 average   |              |            | +4%
```

I was expecting way more, but hey - that's better than nothing.

**Comment by Bensk1** [comment]
I get these results on nemea with clang and a scale factor of 1:

```
 Benchmark | prev. iter/s | new iter/s | change        
-----------+--------------+------------+---------------
 TPC-H 1   | 0.062        | 0.066      | +6%  
 TPC-H 3   | 1.335        | 1.173      | -12% 
 TPC-H 5   | 0.043        | 0.043      | -2%  
 TPC-H 6   | 6.547        | 6.379      | -3%  
 TPC-H 7   | 0.182        | 0.18       | -1%  
 TPC-H 9   | 0.033        | 0.033      | -1%  
 TPC-H 10  | 1.071        | 1.095      | +2%  
 TPC-H 13  | 0.939        | 0.943      | +0%  
 average   |              |            | -1%  
```

**Comment by mrks** [comment]
This is really weird, because I am quite sure that LTO should have a significant benefit. And as we can see with the failing CI build, *something* happens.

Unfortunately, I don't know a way to see if LTO is actually doing its thing:
https://stackoverflow.com/questions/51048414/clang-how-to-check-if-lto-was-performed

**Comment by mrks** [comment]
Gnah. It looks like the JIT code assumes in too many places that every pointer points to at least 64 bits. Looks fixed to me - @FabianWiebe, could you review?

File: src/test/operators/jit_operator/specialization/get_runtime_pointer_for_value_test.cpp [hunk 1]
```diff
@@ -60,6 +60,35 @@ class GetRuntimePointerForValueTest : public BaseTest {
     ASSERT_EQ(known_runtime_pointer->address(), expected_address);
   }
 
+  __attribute__((no_sanitize_address)) void bitcode_pointer_test() {
```
Comment (mrks) [inline src/test/operators/jit_operator/specialization/get_runtime_pointer_for_value_test.cpp]: This is code moved 1:1 from below.

**Pull request merged**